### PR TITLE
CASMINST-3902: Add and update procedures related to adding river cabinets to a system.

### PR DESCRIPTION
**Pull request by rsjostrand-hpe**
# Description

CASMINST-3902: Add and update procedures to add river cabinets to a
system.

CASMHMS-5462: Added a procedure to configure the root user account a HPE
Node. Updated the Gigabyte controller procedure.

Updated the add a standard rack node to reference WLM documentation.

Added some steps to retrieve the current expected SNMP password on sw-leaf-bmc switches.

Reworked the add NCN procedure to have better flow when configuring settings on replacement hardware.
- Added a procedure to enable IPMI access on new HPE nodes, and reference it a better time when the BMC is actually reachable. This is a procedure that can be used for all types of HPE Nodes.
- Configure root user procedures at a time where the BMC is actually reachable.
- Also added a procedure to set time on HPE nodes. This allows the documentation to point toward documentation that is valid after the system is installed, instead of on the PIT node.

# Checklist Before Merging

- [ ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [ ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [ ] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

**Review by shunr-hpe — APPROVED** [review]

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
I haven't finished reviewing this PR yet, but I have left my comments so far on the main branch PR:
https://github.com/Cray-HPE/docs-csm/pull/2344

**Comment by mharding-hpe** [comment]
Left more comments on the main branch PR

**Comment by rsjostrand-hpe** [comment]
I have applied all of the changes from feedback from https://github.com/Cray-HPE/docs-csm/pull/2344 to this PR.

**Comment by mharding-hpe** [comment]
I'm adding the "do not merge" label because I want to review this before we merge it (and have had bad luck with things being merged when I have only commented on this in my feedback but not added the label). This does not imply anything about my feelings as to the quality of the content here -- it just means that I don't know if I will have time to review this before the first time it will be considered for merging.

I will remove the label once I've finished my reviews and issues identified (if any) are addressed.

File: operations/node_management/Add_additional_Air-Cooled_Cabinets_to_a_System.md
Comment (mharding-hpe) [inline operations/node_management/Add_additional_Air-Cooled_Cabinets_to_a_System.md:116]:
```suggestion
    > cray hsm service values subrole list --format toml
```

File: operations/node_management/Add_a_Standard_Rack_Node.md
Comment (mharding-hpe) [inline operations/node_management/Add_a_Standard_Rack_Node.md:41]:
```suggestion
        > cray hsm service values subrole list --format toml
```

File: operations/node_management/Enable_ipmi_access_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/node_management/Enable_ipmi_access_on_HPE_iLO_BMCs.md:35]:
```suggestion
    curl -k -u "root:${ROOT_PASSWORD}" "https://${BMC}/redfish/v1/Managers/1/NetworkProtocol" | jq .IPMI
```

File: operations/node_management/Enable_ipmi_access_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/node_management/Enable_ipmi_access_on_HPE_iLO_BMCs.md:113]:
```suggestion
        curl -k -u "root:${ROOT_PASSWORD}" "https://${BMC}/redfish/v1/Managers/1/NetworkProtocol" | jq .IPMI
```

File: operations/node_management/Update_the_HPE_Node_BIOS_Time.md
Comment (mharding-hpe) [inline operations/node_management/Update_the_HPE_Node_BIOS_Time.md:72]:
```suggestion
    ipmitool -I lanplus -U root -E -H "${BMC}" chassis power off
```

File: operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md:19]:
```suggestion
        EXPECTED_ROOT_PASSWORD=$(cray scsd bmc creds list --targets "${BMC}" --format json | jq .Targets[].Password -r)
```

File: operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md:50]: Is the export necessary? I don't see a case where we're starting some subprocess that needs to access it. I only see it being used on the command lines here.
```suggestion
    BMC=x3000c0s3b0
```

File: operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md:56]: Is the export necessary? I don't see a case where we're starting some subprocess that needs to access it. I only see it being used on the command lines here.
```suggestion
    BMC=10.254.1.9
```

File: operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md:74]:
```suggestion
    curl -k -u admin:"${IPMI_PASSWORD}" "https://${BMC}/redfish/v1/Managers" -i | head -1
```

File: operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md:96]:
```suggestion
    ipmitool -U admin -E -I lanplus -H "${BMC}" user set name 4 root
    ipmitool -U admin -E -I lanplus -H "${BMC}" user set password 4 "${EXPECTED_ROOT_PASSWORD}"
    ipmitool -U admin -E -I lanplus -H "${BMC}" user priv 4 4 1
    ipmitool -U admin -E -I lanplus -H "${BMC}" user enable 4
    ipmitool -U admin -E -I lanplus -H "${BMC channel setaccess 1 4 callin=on ipmi=on link=on
```

File: operations/node_management/Add_additional_Air-Cooled_Cabinets_to_a_System.md
Comment (mharding-hpe) [inline operations/node_management/Add_additional_Air-Cooled_Cabinets_to_a_System.md:132]:
```suggestion
        "registry.local/artifactory.algol60.net/csm-docker/stable/hardware-topology-assistant:${HTA_VERSION}" \
        update "${CCJ_FILE}"
```

File: operations/node_management/Add_additional_Air-Cooled_Cabinets_to_a_System.md
Comment (mharding-hpe) [inline operations/node_management/Add_additional_Air-Cooled_Cabinets_to_a_System.md:151]:
```suggestion
    /usr/share/doc/csm/scripts/operations/node_management/Add_River_Cabinets/update_ncn_etc_hosts.py "${TOPOLOGY_CHANGES_JSON}"
```

File: operations/node_management/Enable_ipmi_access_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/node_management/Enable_ipmi_access_on_HPE_iLO_BMCs.md:64]:
```suggestion
        curl -k -u "root:${ROOT_PASSWORD}" -X PATCH \
            -H 'Content-Type: application/json' \
            -d '{"IPMI": {"Port": 623, "ProtocolEnabled": true}}' \
            "https://${BMC}/redfish/v1/Managers/1/NetworkProtocol" | jq
```

File: operations/node_management/Enable_ipmi_access_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/node_management/Enable_ipmi_access_on_HPE_iLO_BMCs.md:89]:
```suggestion
        curl -k -u "root:${ROOT_PASSWORD}" -X POST \
            -H 'Content-Type: application/json' \
            -d '{"ResetType": "GracefulRestart"}' \
            "https://${BMC}/redfish/v1/Managers/1/Actions/Manager.Reset" | jq
```

File: operations/node_management/Update_the_HPE_Node_BIOS_Time.md
Comment (mharding-hpe) [inline operations/node_management/Update_the_HPE_Node_BIOS_Time.md:60]:
```suggestion
    > ssh -L "9443:${bmc}:443" -N root@eniac-ncn-m001
```

File: operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md:110]:
```suggestion
    ipmitool -U admin -E -I lanplus -H "${BMC}" sol payload enable 1 4
```

File: operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md:118]:
```suggestion
        curl -s -k -u admin:"${IPMI_PASSWORD}" "https://${BMC}/redfish/v1/AccountService/Accounts" | jq ".Members"
```

File: operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md:137]:
```suggestion
        curl -s -k -u admin:"${IPMI_PASSWORD}" "https://${BMC}/redfish/v1/AccountService/Accounts/4" | jq '. | { Name: .Name, UserName: .UserName, RoleId: .RoleId }'
```

File: operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md:153]:
```suggestion
    curl -k -u "root:${EXPECTED_ROOT_PASSWORD}" "https://${BMC}/redfish/v1/Managers" -i  | head -1
```

File: operations/security_and_authentication/Change_SNMP_Credentials_on_Leaf_BMC_Switches.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Change_SNMP_Credentials_on_Leaf_BMC_Switches.md:125]:
```suggestion
    1. Check the logs of the REDS pod for SNMP communication issues.

        In the following command, replace `CRAY_REDS_POD_NAME` with the currently running pod for REDS:
```

File: operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md:24]:
```suggestion
        EXPECTED_ROOT_PASSWORD=$(cray scsd bmc creds list --targets "${BMC}" --format json | jq .Targets[].Password -r)
```

File: operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md:55]: Does this need to be exported, or is it just used in the commands on this page?
```suggestion
        BMC=x3000c0s3b0
```

File: operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md:61]: Does this need to be exported, or is it just used in the commands on this page?

```suggestion
        BMC=10.254.1.9
```

File: operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md:67]:
```suggestion
    curl -k -u "root:${EXPECTED_ROOT_PASSWORD}" "https://${BMC}/redfish/v1/Managers" -i  | head -1
```

File: operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md:106]:
```suggestion
    curl -k -u "${DEFAULT_USERNAME}:${DEFAULT_PASSWORD}" "https://${BMC}/redfish/v1/Managers" -i | head -1
```

File: operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md:131]:
```suggestion
        curl -k -u "${DEFAULT_USERNAME}:${DEFAULT_PASSWORD}" "https://${BMC}/redfish/v1/Managers" -i  | head -1
```

File: operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md:145]:
```suggestion
        for account in $(curl -s -u "${DEFAULT_USERNAME}:${DEFAULT_PASSWORD}" -k "https://${BMC}/redfish/v1/AccountService/Accounts" | jq '.Members[]."@odata.id"' -r); do 
```

File: operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md:147]:
```suggestion
            curl -k -s -u "${DEFAULT_USERNAME}:${DEFAULT_PASSWORD}" -k "https://${BMC}${account}" | jq '. | {Id: .Id, UserName: .UserName, RoleId: .RoleId}' -c
```

File: operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md:173]:
```suggestion
        curl -k -u "${DEFAULT_USERNAME}:${DEFAULT_PASSWORD}" -X POST \
                -H 'Content-Type: application/json' \
                -d $(jq --arg PASSWORD "${EXPECTED_ROOT_PASSWORD}" -nc '{RoleId: "Administrator", UserName: "root", Password: $PASSWORD}') \
                "https://${BMC}/redfish/v1/AccountService/Accounts" | jq
```

File: operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md:233]:
```suggestion
            curl -k -u "${DEFAULT_USERNAME}:${DEFAULT_PASSWORD}" -X PATCH \
                -H 'Content-Type: application/json' \
                -d $(jq --arg PASSWORD "${EXPECTED_ROOT_PASSWORD}" -nc '{Password: $PASSWORD}') \
                "https://${BMC}/redfish/v1/AccountService/Accounts/${ROOT_USER_ACCOUNT_ID}" | jq
```

File: operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md
Comment (mharding-hpe) [inline operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md:255]:
```suggestion
    curl -k -u "root:${EXPECTED_ROOT_PASSWORD}" "https://${BMC}/redfish/v1/Managers" -i  | head -1
```

**Review by mharding-hpe — CHANGES_REQUESTED** [review]

**Comment by rsjostrand-hpe** [comment]
@mharding-hpe I've applied your feedback.

**Review by mharding-hpe — APPROVED** [review]

**Review by rustydb — APPROVED** [review]

**Pull request merged**
